lib/chewy/type/adapter/active_record.rb

```diff
@@ -4,6 +4,44 @@ module Chewy
   class Type
     module Adapter
       class ActiveRecord < Orm
+        IMPORT_OVERHEAD_LIMIT = 2
+
+        class InfiniteImportWarning < RuntimeError
+          def initialize(batch_size)
+            @batch_size = batch_size
+          end
+
+          def message
+            <<-MSG.strip_heredoc
+              Import process indexed already #{IMPORT_OVERHEAD_LIMIT}x
+              as many object as there were in database when import
+              started. It probably means that you are creating or
+              updating more objects that are indexed in one batch
+              during time batch get processed.
+
+              Possible solutions:
+               - Increase batch size (current size: #{@batch_size})
+               - Boost batch import by making index entry generation faster
+                 (e.g. check if your type definition do not have N+1 query
+                 problem)
+            MSG
+          end
+        end
+
+        class InfiniteLoopIndicator
```

Comment:
`InfiniteLoopDetector` maybe? I have no strong opinion on that, it just feels more accurate (to me).